### PR TITLE
added hexapad a custom handwired hexadecimal number pad

### PR DESCRIPTION
**Pull request by samux6146**
## Description

i added a keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by samux6146** [comment]
i don't understand why lint makes me errors

**Comment by waffle87** [comment]
Read the warnings when compiling:
```
Generating: .build/obj_hexapad/src/default_keyboard.h                                               [WARNINGS]
 |
 | ☒ hexapad: Layout "LAYOUT" has no "matrix" definition in either "info.json" or "<keyboard>.h"!
 | ☒ hexapad: No LAYOUTs defined! Need at least one layout defined in info.json.
 |
```
(hint `KEYMAP` layout macros are deprecate)

**Review by elpekenin — COMMENTED** [review]
You shouldn't have any changes to `/lib`
Check that your local copy of the repo is (somewhat) up to date with qmk/qmk_firmware and run `make git-submodule`

File: keyboards/handwired/hexapad/config.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/config.h:17]:
```suggestion
#pragma once
```

File: keyboards/handwired/hexapad/config.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+#include "config_common.h"
+
+
+/* key matrix size */
+#define MATRIX_ROWS 7
+#define MATRIX_COLS 4
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/config.h:24]: Im not 100% sure about these
```suggestion
```

File: keyboards/handwired/hexapad/config.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/config.h:40]: Default value, no need to re-define it
```suggestion
```

File: keyboards/handwired/hexapad/config.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+#endif
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/config.h:53]: `#pragma once`-related adjustment
```suggestion
```

File: keyboards/handwired/hexapad/hexapad.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef HEXAPAD_H
+#define HEXAPAD_H
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/hexapad.h:17]:
```suggestion
#pragma once
```

File: keyboards/handwired/hexapad/hexapad.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+#include "quantum.h"
+
+#define KEYMAP( \
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/hexapad.h:21]:
```suggestion
#define LAYOUT( \
```

File: keyboards/handwired/hexapad/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+	KEYMAP(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS)
+
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/keymaps/default/keymap.c:163]: Useless block of empty layers
```suggestion
```

File: keyboards/handwired/hexapad/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS),
+
+[3] = KEYMAP(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS),
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/keymaps/via/keymap.c:54]: Im not a VIA user, but i think it works fine without the need to add empty layers until you have 4 of them
```suggestion
```

**Review by elpekenin — CHANGES_REQUESTED** [review]
Some of these changes (and probably some more that i've missed) are in the [PR Checklist](https://docs.qmk.fm/#/pr_checklist)

File: keyboards/handwired/hexapad/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "hexapad.h"
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	KEYMAP(
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/keymaps/default/keymap.c:20]:
```suggestion
	[0] = LAYOUT(
```

File: keyboards/handwired/hexapad/hexapad.h [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+	{ K00,   K01,   K02,   K03 }, \
+	{ K10,   K11,   K12,   K13 }, \
+	{ K20,   K21,   K22,   K23 }, \
+	{ K30,   K31,   K32,   KC_NO }, \
+	{ K40,   K41,   K42,   K43 }, \
+	{ K50,   K51,   K52,   KC_NO }, \
+	{ KC_NO, K61,   K62,   K63 }  \
+}
+
+#endif
```
Comment (elpekenin) [inline keyboards/handwired/hexapad/hexapad.h:39]: `#pragma once` again
```suggestion
```

**Pull request closed without merging**